pyproject.toml

```diff
@@ -20,6 +20,21 @@ license = { text = "BSD 2-clause" }
 requires-python = ">=3.9"
 dependencies = ["numba>=0.59.1"]
 
+[project.optional-dependencies]
+cu12 = [
+    "cuda-python==12.9.*",  # supports all CTK 12.x
+    "nvidia-cuda-nvcc-cu12",  # for libNVVM
+    "nvidia-cuda-runtime-cu12",
+    "nvidia-cuda-nvrtc-cu12",
+]
+test = [
+    "psutil",
+    "cffi",
+    "nvidia-curand-cu12",
+    "pynvjitlink-cu12",
```

Comment:
What if someone wants to test cu11?

Reply:
Good idea. I split out `test-cu12` from `test` in commit 4dad6c25. This does make the CI scripts slightly more general.